--- a/dlls/winewayland.drv/waylanddrv.h
+++ b/dlls/winewayland.drv/waylanddrv.h
@@ -41,6 +41,7 @@
 #include "pointer-warp-v1-client-protocol.h"
 #include "alpha-modifier-v1-client-protocol.h"
 #include "fractional-scale-v1-client-protocol.h"
+#include "linux-dmabuf-unstable-v1-client-protocol.h"
 
 #include "windef.h"
 #include "winbase.h"
@@ -185,6 +186,7 @@
     struct wp_cursor_shape_manager_v1 *wp_cursor_shape_manager_v1;
     struct wp_pointer_warp_v1 *wp_pointer_warp_v1;
     struct wp_alpha_modifier_v1 *wp_alpha_modifier_v1;
+    struct zwp_linux_dmabuf_v1 *zwp_linux_dmabuf_v1;
     struct wayland_seat seat;
     struct wayland_keyboard keyboard;
     struct wayland_pointer pointer;
@@ -300,6 +302,46 @@
 
 BOOL wayland_process_init(void);
 
+/**********************************************************************
+ *          Wayland dmabuf buffer
+ */
+
+#define WINEWAYLAND_MAX_DMABUF_PLANES 4
+
+typedef void (*dmabuf_buffer_release_cb)(struct wayland_dmabuf_buffer *buffer);
+
+struct wayland_dmabuf_buffer
+{
+    struct wl_buffer *wl_buffer;
+    int width, height;
+    uint32_t format;
+    uint64_t modifier;
+    int plane_count;
+    int fds[WINEWAYLAND_MAX_DMABUF_PLANES];
+    uint32_t offsets[WINEWAYLAND_MAX_DMABUF_PLANES];
+    uint32_t strides[WINEWAYLAND_MAX_DMABUF_PLANES];
+    dmabuf_buffer_release_cb release_callback;
+    BOOL busy;
+};
+
+BOOL wayland_dmabuf_init(struct wl_registry *registry, uint32_t id, uint32_t version);
+BOOL wayland_dmabuf_has_support(void);
+struct wayland_dmabuf_buffer *wayland_dmabuf_buffer_create(
+    int width, int height, uint32_t format, uint64_t modifier,
+    int plane_count, const int *fds, const uint32_t *offsets,
+    const uint32_t *strides, dmabuf_buffer_release_cb release_cb);
+struct wayland_dmabuf_buffer *wayland_dmabuf_wl_buffer_for_gpu_fd(
+    int width, int height, int gpu_fd, uint32_t gpu_format,
+    uint64_t gpu_modifier, uint32_t gpu_stride,
+    dmabuf_buffer_release_cb release_cb);
+void wayland_dmabuf_buffer_destroy(struct wayland_dmabuf_buffer *buffer);
+
 /**********************************************************************
  *          Wayland output
  */
@@ -355,6 +397,9 @@
 void wayland_surface_attach_shm(struct wayland_surface *surface,
                                 struct wayland_shm_buffer *shm_buffer,
                                 HRGN surface_damage_region);
+void wayland_surface_attach_dmabuf(struct wayland_surface *surface,
+                                   struct wayland_dmabuf_buffer *dmabuf_buffer,
+                                   HRGN surface_damage_region);
 BOOL wayland_surface_reconfigure(struct wayland_surface *surface);
 BOOL wayland_surface_config_is_compatible(struct wayland_surface_config *conf, RECT rect,
                                           enum wayland_surface_config_state state);